tensorflow_app/views/import_graphdef.py

```diff
@@ -262,6 +292,28 @@ def import_graph_def(request):
                         return JsonResponse({'result': 'error',
                                              'error': 'Missing shape info in GraphDef'})
 
+                if str(node.type) == 'AvgPool3D':
+                    layer['params']['pool'] = 'AVE'
+                    layer['params']['kernel_d'] = int(
+                        node.get_attr('ksize')[1])
+                    layer['params']['kernel_h'] = int(
+                        node.get_attr('ksize')[2])
+                    layer['params']['kernel_w'] = int(
+                        node.get_attr('ksize')[3])
+                    layer['params']['stride_d'] = int(
+                        node.get_attr('strides')[1])
+                    layer['params']['stride_h'] = int(
+                        node.get_attr('strides')[2])
+                    layer['params']['stride_w'] = int(
+                        node.get_attr('strides')[3])
+                    layer['params']['layer_type'] = '3D'
+                    try:
+                        layer['params']['pad_d'], layer['params']['pad_h'], \
+                            layer['params']['pad_w'] = get_padding(node, layer)
+                    except TypeError:
+                        return JsonResponse({'result': 'error', 'error':
+                                             'Missing shape info in GraphDef'})
+
```

Comment:
These lines are duplicated in `MaxPool3D`. You can try to make it hierarchical -- first, check if `node.type` has a substring`3D` and then check if it is max or average. That should reduce a lot of duplication.

Reply:
Similar comment goes for `MaxPool` and `AvgPool` variants.